### PR TITLE
[eiger] Matplotlib and dependencies on Eiger

### PR DESCRIPTION
**Pull request by lucamar**
Updated recipes for Jupyterlab and dependencies on Eiger.

### CODE REVIEW

**Comment by lucamar** [comment]
I have provided `matplotlib` with its dependencies as a separate modulefile: practically, it replaces the old `PyExtensions`, since most of the other Python extensions are now available with `cray-python`, although the version is not the latest. The extensions are built, but I used to get errors from the last line in `postinstallcmds`. 

I have questions on the way to deal with the packages `dask` and `ipcmagic` that may be addressed offline.

**Comment by lucamar** [comment]
Log file:  `/scratch/e1000/jenscscs/TestingEB/eiger/27d38fc-4356/tmp/eb-O9uHsK/easybuild-jupyterlab-2.2.8-20201112.225610.tyVXt.log`

**Comment by lucamar** [comment]
@rsarm I left in the pull requests some dependencies of `Jupyterlab`, although I have removed the recipe `jupyterlab` that will be created by @twrobinson in a separate pull request. 
The main novelty in this pull request is the `matplotlib` recipe replacing `PyExtensions`: please let me know your opinion about it.

**Comment by lucamar** [comment]
The recipe `graphviz-2.44.1-cpeGNU-20.10.eb` failed to download the tarball: however, it must have been a temporary issue, since the manual download works from the compute nodes of Eiger: 
```
$ srun --pty /bin/bash

$ hostname
nid001010

$ curl https://www2.graphviz.org/Packages/stable/portable_source/graphviz-2.44.1.tar.gz --output graphviz-2.44.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 32.5M  100 32.5M    0     0  9797k      0  0:00:03  0:00:03 --:--:-- 9797k

$ ls -ltrh graphviz-2.44.1.tar.gz
-rw-r--r-- 1 lucamar csstaff 33M Nov 17 13:48 graphviz-2.44.1.tar.gz
```

**Comment by lucamar** [comment]
Also the manual build of `graphviz-2.44.1-cpeGNU-20.10.eb` on compute nodes works: 
```
$ srun -u eb graphviz-2.44.1-cpeGNU-20.10.eb --ignore-locks -rf
== temporary log file in case of crash /tmp/lucamar/eb-ESnAaB/easybuild-ZPtPHL.log
== resolving dependencies ...
== processing EasyBuild easyconfig /users/lucamar/GitHub/lucamar/production/easybuild/easyconfigs/g/graphviz/graphviz-2.44.1-cpeGNU-20.10.eb
== building and installing graphviz/2.44.1-cpeGNU-20.10...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== taking care of extensions...
== restore after iterating...
== postprocessing...
== sanity checking...
== cleaning up...
== creating module...
== permissions...
== packaging...
== COMPLETED: Installation ended successfully (took 2 min 14 sec)
== Results of the build can be found in the log file(s) /users/lucamar/easybuild/eiger/software/graphviz/2.44.1-cpeGNU-20.10/easybuild/easybuild-graphviz-2.44.1-20201117.135927.log
== Build succeeded for 1 out of 1
== Temporary log file(s) /tmp/lucamar/eb-ESnAaB/easybuild-ZPtPHL.log* have been removed.
== Temporary directory /tmp/lucamar/eb-ESnAaB has been removed.
```

**Comment by lucamar** [comment]
retest this please

File: easybuild/easyconfigs/m/matplotlib/matplotlib-3.3.3-cpeGNU-20.10.eb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+homepage = 'https://matplotlib.org'
+description = """matplotlib is a python 2D plotting library which produces publication quality figures in a variety of
+ hardcopy formats and interactive environments across platforms. matplotlib can be used in python scripts, the python
+ and ipython shell, web application servers, and six graphical user interface toolkits."""
+
+toolchain = {'name': 'cpeGNU', 'version': '20.10'}
+toolchainopts = {'pic': True, 'verbose': False}
+
+dependencies = [
+    ('cray-mpich', EXTERNAL_MODULE),
```
Comment (rsarm) [inline easybuild/easyconfigs/m/matplotlib/matplotlib-3.3.3-cpeGNU-20.10.eb:16]: Is `cray-mpich` needed here?

Comment (lucamar) [reply easybuild/easyconfigs/m/matplotlib/matplotlib-3.3.3-cpeGNU-20.10.eb:16]: Probably not, as far as I can see on https://matplotlib.org/users/installing.html. I can remove the dependency

**Comment by lucamar** [comment]
Now `libsodium-1.0.18-cpeGNU-20.10.eb` failed to download... again a temporary issue, I assume, since it succeeded before.

**Comment by lucamar** [comment]
retest this please

**Review by rsarm — APPROVED** [review]

**Pull request merged**